app/src/main/java/com/mk/tasky/agenda/presentation/detail/components/DetailPhotoSelector.kt

```diff
@@ -0,0 +1,193 @@
+import com.mk.tasky.ui.theme.Light2
+import com.mk.tasky.ui.theme.LightBlue
+
+@OptIn(ExperimentalPermissionsApi::class)
+@Composable
+fun DetailPhotoSelector(
+    photos: List<AgendaPhoto>,
+    onPhotoSelected: (Uri) -> Unit,
+    modifier: Modifier = Modifier
+) {
+    val galleryLauncher =
+        rememberLauncherForActivityResult(ActivityResultContracts.PickVisualMedia()) {
+            if (it != null) {
+                onPhotoSelected(it)
+            }
+        }
+
+    val readGalleryPermissionState =
+        rememberPermissionState(Manifest.permission.READ_MEDIA_IMAGES)
+
+    Row(
+        modifier = modifier
+            .fillMaxWidth()
+            .background(color = Light2)
+    ) {
+        if (photos.isEmpty()) {
+            EmptyPhotoPicker(onAddPhoto = {
+                launchImageGallery(readGalleryPermissionState, galleryLauncher)
+            })
+        } else {
+            PhotoViewer(photos = photos, onPhotoClick = { }, onAddPhoto = {
+                launchImageGallery(readGalleryPermissionState, galleryLauncher)
+            })
+        }
+    }
+}
+
+@OptIn(ExperimentalPermissionsApi::class)
+private fun launchImageGallery(
+    permissionState: PermissionState,
+    galleryLauncher: ManagedActivityResultLauncher<PickVisualMediaRequest, Uri?>
+) {
+    if (permissionState.status == PermissionStatus.Granted || BuildConfig.DEBUG) { // Emulator doesn't run the permission checker
+        val mediaRequest =
+            PickVisualMediaRequest(ActivityResultContracts.PickVisualMedia.ImageOnly)
+        galleryLauncher.launch(mediaRequest)
+    } else {
+        permissionState.launchPermissionRequest()
+    }
+}
+
+@Composable
+private fun PhotoViewer(
+    photos: List<AgendaPhoto>,
+    onPhotoClick: (AgendaPhoto) -> Unit,
+    onAddPhoto: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    Column(modifier = modifier.padding(21.dp)) {
+        Text(
+            text = stringResource(R.string.photos),
+            color = Black,
+            fontWeight = FontWeight.SemiBold,
+            fontSize = 20.sp
+        )
+        Spacer(modifier = Modifier.height(21.dp))
+        LazyRow(modifier = Modifier.fillMaxWidth()) {
+            items(photos) {
+                Box(
+                    modifier = Modifier
+                        .size(60.dp)
+                        .clickable {
+                            onPhotoClick(it)
+                        }
+                        .border(
+                            BorderStroke(2.dp, color = LightBlue),
+                            shape = RoundedCornerShape(5.dp)
+                        ),
+                    contentAlignment = Alignment.Center
+                ) {
+                    AsyncImage(
+                        model = ImageRequest.Builder(LocalContext.current)
+                            .data(it.location)
+                            .crossfade(true)
+                            .build(),
+                        contentDescription = "",
```

Comment:
I'd prefer to set it to `null` if it doesn't exist, since that's what null values are used for :)